chapter05/ajd/exercise2.pl

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+if ( !scalar @ARGV ) {
+  die "usage: $0 <some data file>\n";
+}
+
+# Get the file read
+my %total_bytes;
+
+while ( <> ) {
+  if ( /^#/) { next };
+  my ( $source, $destination, $bytes ) = split;
+  $total_bytes{$source}{$destination} += $bytes;
+}
+
+# Sorting and printing
+foreach my $host ( sort { by_host_data( $a, $b, \%total_bytes ) } keys %total_bytes ) {
+
+  foreach my $dest ( sort { by_dest_data( $a, $b, $total_bytes{$host} ) } keys $total_bytes{$host} ) {
```

Comment:
The body of the sort would probably be clearer _without_ the function call in this case.